src/tests/testClpMap.cc

```diff
@@ -0,0 +1,133 @@
+        m.add("foo", 0);
+        CPPUNIT_ASSERT_EQUAL(static_cast<size_t>(11), m.entries());
+    }
+    {
+        TestMap m(1024, 5);
+        addSequenceOfElementsToMap(m, 1000, 0, 10);
+        CPPUNIT_ASSERT(m.entries() < 1000);
+    }
+}
+
+void
+testClpMap::testMemoryCounter()
+{
+    CPPUNIT_ASSERT_EQUAL(sizeof(int), static_cast<size_t>(DefaultMemoryUsage(int())));
+    CPPUNIT_ASSERT_EQUAL(sizeof(int32_t), static_cast<size_t>(DefaultMemoryUsage(int32_t())));
+    CPPUNIT_ASSERT_EQUAL(sizeof(int64_t), static_cast<size_t>(DefaultMemoryUsage(int64_t())));
+    CPPUNIT_ASSERT_EQUAL(sizeof(char), static_cast<size_t>(DefaultMemoryUsage(char())));
+    char str[10];
+    CPPUNIT_ASSERT_EQUAL(sizeof(str), static_cast<size_t>(DefaultMemoryUsage(str)));
+    CPPUNIT_ASSERT_EQUAL(sizeof(std::string), static_cast<size_t>(DefaultMemoryUsage(std::string())));
+}
+
+void
+testClpMap::testConstructor()
+{
+    const TestMap nilA(0);
+    CPPUNIT_ASSERT_EQUAL(uint64_t(0), nilA.memLimit());
+    CPPUNIT_ASSERT_EQUAL(uint64_t(0), nilA.freeMem());
+    CPPUNIT_ASSERT_EQUAL(uint64_t(0), nilA.memoryUsed());
+    CPPUNIT_ASSERT_EQUAL(size_t(0), nilA.entries());
+
+    const TestMap nilB(0, 0);
+    CPPUNIT_ASSERT_EQUAL(uint64_t(0), nilB.memLimit());
+    CPPUNIT_ASSERT_EQUAL(uint64_t(0), nilB.freeMem());
+    CPPUNIT_ASSERT_EQUAL(uint64_t(0), nilB.memoryUsed());
+    CPPUNIT_ASSERT_EQUAL(size_t(0), nilB.entries());
+
+    const TestMap emptyC(1);
+    CPPUNIT_ASSERT_EQUAL(uint64_t(1), emptyC.memLimit());
+    CPPUNIT_ASSERT_EQUAL(uint64_t(1), emptyC.freeMem());
+    CPPUNIT_ASSERT_EQUAL(uint64_t(0), emptyC.memoryUsed());
+    CPPUNIT_ASSERT_EQUAL(size_t(0), emptyC.entries());
+
+    const TestMap emptyD(1024);
+    CPPUNIT_ASSERT_EQUAL(uint64_t(1024), emptyD.memLimit());
+    CPPUNIT_ASSERT_EQUAL(uint64_t(1024), emptyD.freeMem());
+    CPPUNIT_ASSERT_EQUAL(uint64_t(0), emptyD.memoryUsed());
+    CPPUNIT_ASSERT_EQUAL(size_t(0), emptyD.entries());
+}
+
+void
+testClpMap::testSetMemLimit()
+{
+    TestMap m(2048);
+    // overflow the map with entries to make sure it has lots of entries to purge below
+    addSequenceOfElementsToMap(m, 1000, 0, 10);
+    const auto testEntriesBefore = m.entries();
+    CPPUNIT_ASSERT(testEntriesBefore > 0);
+    m.setMemLimit(m.memoryUsed() / 2);
+    CPPUNIT_ASSERT(testEntriesBefore > m.entries());
+
+    const auto entriesAfterPurge = m.entries(); // TODO: Move and use in the above assertion
+    m.setMemLimit(m.memLimit() * 2);
+    // overflow the map with entries again to make sure it can grow after purging
+    addSequenceOfElementsToMap(m, m.memLimit() / sizeof(int), 0, 10);  // TODO: Extract into overflowMap()
+    CPPUNIT_ASSERT(entriesAfterPurge < m.entries());
+}
+
+void
+testClpMap::testTtlExpiration()
+{
+    TestMap m(2048);
+    m.add(std::to_string(1), 1, 10);
+    CPPUNIT_ASSERT_EQUAL(static_cast<size_t>(1), m.entries());
+    CPPUNIT_ASSERT(static_cast<const int *>(nullptr)!=m.get("1"));
```

Comment:
I hope the simple expression below works. Please let me know if I missed an earlier comment about it.

```suggestion
    CPPUNIT_ASSERT(m.get("1"));
```

Same for other similar occurrences, of course.